surface_reconstruction/nksr/nksr_fvdb/coord_xform.py

```diff
@@ -0,0 +1,516 @@
+        ref_distance = math.sqrt(3.0)
+
+        # Pseudo scaling factor is the ratio
+        return target_distance / ref_distance
+
+    def compose(self, other: "CoordXform") -> "CoordXform":
+        """Compose this transformation with another.
+
+        The resulting transformation applies other first, then self.
+        i.e., (self.compose(other))(x) == self(other(x))
+
+        Args:
+            other: The transformation to apply before this one.
+
+        Returns:
+            A new CoordXform representing the composition.
+        """
+        return ComposedXform(first=other, second=self)
+
+    @overload
+    def __matmul__(self, other: "CoordXform") -> "CoordXform": ...
+
+    @overload
+    def __matmul__(self, other: torch.Tensor) -> torch.Tensor: ...
+    @overload
+    def __matmul__(self, other: fvdb.JaggedTensor) -> fvdb.JaggedTensor: ...
+
+    def __matmul__(
+        self, other: "CoordXform | torch.Tensor | fvdb.JaggedTensor"
+    ) -> "CoordXform | torch.Tensor | fvdb.JaggedTensor":
+        """Matrix-multiply style operator supporting both composition and application.
+
+        This operator is overloaded to support two distinct use cases:
+
+        1. **Composition** (when other is a CoordXform):
+           Returns a new transform representing self(other(x)).
+           Uses mathematical convention where (A @ B)(x) = A(B(x)),
+           so B (other) is applied first, then A (self).
+
+           Example:
+               world_T_object = world_T_camera @ camera_T_object
+               # Equivalent to: world_T_camera.compose(camera_T_object)
+
+        2. **Application** (when other is a Tensor or JaggedTensor):
+           Applies this transform to the coordinates, returning transformed coords.
+           This provides a convenient notation: xform @ coords instead of xform(coords).
+
+           Example:
+               world_coords = world_T_object @ object_coords
+               # Equivalent to: world_T_object.apply(object_coords)
+
+        Args:
+            other: Either a CoordXform (for composition) or coordinates to transform
+                   (torch.Tensor of shape [N, 3] or fvdb.JaggedTensor).
+
+        Returns:
+            - If other is CoordXform: A new CoordXform representing the composition.
+            - If other is torch.Tensor: Transformed coordinates as torch.Tensor.
+            - If other is fvdb.JaggedTensor: Transformed coordinates as fvdb.JaggedTensor.
+
+        Raises:
+            ValueError: If other is not a supported type.
+        """
+        if isinstance(other, torch.Tensor):
+            return self.apply_tensor(other)
+        elif isinstance(other, fvdb.JaggedTensor):
+            return self.apply_jagged(other)
+        elif isinstance(other, CoordXform):
+            return self.compose(other)
+        else:
+            raise ValueError(f"Unsupported type: {type(other)}")
+
+
+@dataclass(frozen=True)
+class ComposedXform(CoordXform):
```

Comment:
Do we need an explicit binary composition? Does it make sense to do something like `torch.nn.Sequential` or `fvdb_reality_capture.transforms.Compose` where the class takes a list of transforms to apply in sequence?